### ISSUE TITLE
Handle nested xml dsl

### ISSUE THREAD

**Issue by AlanFoster** [AlanFoster/Camelry#19]
In a lot of places Camel supports the ability to redefine the core camel route dslelements within EIP components; Currently we do not support this generic concept.

For instance

``` xml
        <route>
            <from uri="..."/>

            <split shareUnitOfWork="true">
                <xpath>/items/item</xpath>
                <!--
                      The route definition can be used within this element now,
                       and should perform the same level of IDE support as
                       outside of the split element
                 -->
                <to uri="vdsm:processor" />
            </split>

        </route>
```

**Comment by AlanFoster** [comment]
I've found that there is a nifty feature in IntelliJ to generate DomElement interfaces from an XSD schema by setting the jvm arg `-Didea.is.internal=true` within `Tools`> Internal Actions`

This might save a lot of time in generating the initial DomElements, and there's the added benefit of having Class names which closely relate to the core Camel codebase too :)

**Comment by AlanFoster** [comment]
There should be enough tests added under #17 to go ahead and try to progress with this issue by generating the DomElement classes, and to start moving over the existing annotation converters to the new classes etc.

**Comment by AlanFoster** [comment]
As an update to this; I proceeded with generating the DomElement interfaces from the camel XSDs, but due to the duplication within the generated schema it was a bit awkward to deal with.

I'm not 100% sure of the best way to deal with this, but i'm sure there's a good solution :)

**Comment by AlanFoster** [comment]
The support for this is much improved now; I have created a basic JAXB Codegen module which will introspect the camel jaxb.index files on the classpath, using the JAXB API, and generate the required DomElements
 files using Scalate, a templating language. 

There was a small manual process involved after the creation of these classes; For instance to include the additional `@Converter` and `@Referencing` annotations etc.

I would definitely like to revisit the Scala code when I've got more experience in FP, to write it in a more idiomatic way in the future.
